crates/task-impls/src/network.rs

```diff
@@ -584,13 +605,14 @@ impl<
             }
             HotShotEvent::ViewChange(view) => {
                 self.view = view;
-                self.network
-                    .update_view::<TYPES>(
-                        self.view.u64(),
-                        self.epoch.u64(),
-                        &self.quorum_membership,
-                    )
-                    .await;
+                // self.cancel_tasks(TYPES::View::new(view.saturating_sub(2)));
+                let net = Arc::clone(&self.network);
+                let epoch = self.epoch.u64();
+                let mem = self.quorum_membership.clone();
+                async_spawn(async move {
+                    net.update_view::<TYPES>(view.saturating_sub(2), epoch, &mem)
```

Comment:
I think I left some stuff in from debugging, trying to make sure we weren't cancelling stuff too early.  This can probably just be view (or at works view -1)